File: turnstile-frontend/src/app/models/dto/transaction-dto.ts
import {UserDTO} from './user-dto';
import {TurnstileDTO} from './turnstile-dto';

export class TransactionDTO {
    id: number;
   date: string;
   time: string;
   current_state : string;
   user: UserDTO;
   turnstile: TurnstileDTO;

   constructor(id: number, date: string, time: string, current_state: string,  user: UserDTO, turnstile: TurnstileDTO) {
     this.id = id;
     this.date = date;
     this.time = time;
     this.current_state = current_state;
     this.user = user;
     this.turnstile = turnstile;

   }
}

File: turnstile-backend/src/main/java/it/tdgc/turnstile/dto/TransactionDTO.java
package it.tdgc.turnstile.dto;

import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.springframework.stereotype.Component;

import java.time.LocalDate;
import java.time.LocalTime;

@Getter
@Setter
@Component
@AllArgsConstructor
@NoArgsConstructor
public class TransactionDTO {
    private Integer id;
    private LocalDate date;
    private LocalTime time;
    private String current_state;
    private UserDTO user;
    private TurnstileDTO turnstile;
}
